fake.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// FakeCallArgs holds all the args passed in a call.
+type FakeCallArgs map[string]interface{}
```

Comment:
I don't think I'd bother naming this type, since it doesn't actually get used as a type anywhere.  Just use map[string]interface{}.